pkg/controller/core/workload_controller.go

```diff
@@ -801,70 +874,90 @@ func (h *resourceUpdatesHandler) queueReconcileForPending(ctx context.Context, _
 	}
 }
 
-type workloadCqHandler struct {
-	client client.Client
+type workloadQueueHandler struct {
+	r *WorkloadReconciler
 }
 
-var _ handler.EventHandler = (*workloadCqHandler)(nil)
+var _ handler.EventHandler = (*workloadQueueHandler)(nil)
 
 // Create is called in response to a create event.
-func (w *workloadCqHandler) Create(ctx context.Context, ev event.CreateEvent, wq workqueue.RateLimitingInterface) {
-	if cq, isQueue := ev.Object.(*kueue.ClusterQueue); isQueue {
-		w.queueReconcileForWorkloads(ctx, cq.Name, wq)
+func (w *workloadQueueHandler) Create(ctx context.Context, ev event.CreateEvent, wq workqueue.RateLimitingInterface) {
+	if cq, isCq := ev.Object.(*kueue.ClusterQueue); isCq {
+		w.queueReconcileForWorkloadsOfClusterQueue(ctx, cq.Name, wq)
+	}
+	if lq, isLq := ev.Object.(*kueue.LocalQueue); isLq {
```

Comment:
`else if` or `return` after 886, the object cannot be both cq and lq

Reply:
Done